Headers/Foundation/NSFileManager.h

```diff
@@ -659,6 +669,68 @@ GS_EXPORT NSString* const NSFileSystemFreeNodes;
 - (NSUInteger) fileSystemFileNumber;
 @end
 
+@protocol NSFileManagerDelegate <NSObject>
```

Comment:
Please have a look at NSMetaData.h on how to do this in a way old gcc can cope with.

@protocol NSMetadataQueryDelegate
#if OS_API_VERSION(MAC_OS_X_VERSION_10_6,GS_API_LATEST) && GS_PROTOCOLS_HAVE_OPTIONAL
@optional
#else
@end
@interface NSObject (NSMetadataQueryDelegate)
#endif // GS_PROTOCOLS_HAVE_OPTIONAL